### PR TITLE
add support for ccp only upgrade and default to upgrade all

### PR DESCRIPTION
**Pull request by xizhamsft**
With the new 0801 api version, we now supports control plane only cluster upgrade. Added the control-plane-only argument for this purpose and change the default to upgrade all (by setting kubernetes version of all agentpool profiles explicitly).

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -1112,8 +1113,43 @@ def aks_upgrade(cmd, client, resource_group_name, name, kubernetes_version, no_w
...
                            "attempt resolution of failed cluster state.", instance.kubernetes_version)
 
+    from knack.prompting import prompt_y_n
+
+    upgrade_all = False
     instance.kubernetes_version = kubernetes_version
 
+    vmas_cluster = False
+    for agent_profile in instance.agent_pool_profiles:
+        if agent_profile.type == "AvailabilitySet":
```
Comment (xuto2) [inline src/aks-preview/azext_aks_preview/custom.py]: case insensitive compare

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -1112,8 +1113,43 @@ def aks_upgrade(cmd, client, resource_group_name, name, kubernetes_version, no_w
...
+            vmas_cluster = True
+            print("VMAS cluster")
```
Comment (xuto2) [inline src/aks-preview/azext_aks_preview/custom.py]: remove

**Comment by xuto2** [comment]
check https://github.com/Azure/azure-cli/pull/10513/files#diff-fe2d1dd3c9fc1d841b3437f1a9aac146 for main CLI sync

**Comment by zikalino** [comment]
ok, decided to merge it:
actually I was wondering if we could handle following scenarios, when user specified:

--control-plane-only

and
--control-plane-only False

specifically to avoid displaying prompt.

But this can be considered / checked later.

**Pull request merged**